src/syntax_highlight.rs

```diff
@@ -41,33 +39,33 @@ lazy_static!{
 struct CodeBlock {
     lang: Option<String>,
     code: String,
+    theme: Theme,
 }
 
 impl Renderable for CodeBlock {
     fn render(&self, _: &mut Context) -> Result<Option<String>, Error> {
-
         let syntax = match self.lang {
                 Some(ref lang) => SETUP.syntax_set.find_syntax_by_token(lang),
                 _ => None,
             }
             .unwrap_or_else(|| SETUP.syntax_set.find_syntax_plain_text());
 
-        Ok(Some(highlighted_snippet_for_string(&self.code,
-                                               syntax,
-                                               &SETUP.theme_set.themes[THEME_NAME])))
+        Ok(Some(highlighted_snippet_for_string(&self.code, syntax, &self.theme)))
     }
 }
 
 pub struct DecoratedParser<'a> {
     h: Option<HighlightLines<'a>>,
     parser: Parser<'a>,
+    config: &'a Config,
 }
 
 impl<'a> DecoratedParser<'a> {
-    pub fn new(parser: Parser<'a>) -> Self {
+    pub fn new(parser: Parser<'a>, config: &'a Config) -> Self {
```

Comment:
I'm a little iffy on `Config` being passed everywhere.

It starts to turn it into a god object which will make unit testing more difficult.  I know it saved a little on parameter lists with `excerpt_separator` but in that case I'm actually removing that in change I'm preparing to post.